### PR TITLE
Places

### PR DESCRIPTION
**Pull request by angelk**
Add functionality to add places

- [x] ACL
- [x] translations

### CODE REVIEW

File: src/Controller/Home.php [hunk 1]
```diff
@@ -90,15 +116,58 @@ public function find($neLat, $neLon, $swLat, $swLon, Request $request, TrackRepo
...
+        $data = $qb->select($qb->expr()->count('g.id'))
+            ->getQuery()
+            ->getSingleResult();
+
+        $count = current($data);
+
+        if ($count > 10) {
+            $status = 2; // 2 = too many tracks
+        } else {
+            $status = 1; // 1 = ok
```
[the remark this answers is not in the document]
Comment (angelk) [reply src/Controller/Home.php]: Added ApiResponseInterface

File: src/Entity/Place.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ * @ORM\Table(name="place")
+ */
+class Place implements CreatedAtInterface
+{
+    use PointTrait;
+    use NameTrait;
+    use SendByTrait;
+
+    public const TYPE_GENERIC = 0;
+    public const TYPE_WATER = 1;
```
[the remark this answers is not in the document]
Comment (angelk) [reply src/Entity/Place.php]: Renamed to `TYPE_DRINKING_FOUNTAIN`

File: src/Contract/Http/ApiResponseInterface.php [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace App\Contract\Http;
+
+class ApiResponseInterface
+{
+    public const STATUS_OK = 1;
+    public const STATUS_TOO_MANY_ROWS_FOUND = 2;
```
Comment (StefanMavrodiev) [inline src/Contract/Http/ApiResponseInterface.php]: Usually OK status is 0.

File: src/Upload/ImageUploader.php [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        }
+
+        $file = $request->files->get('file');
+        /* @var $file UploadedFile */
+        if (!$file->isValid()) {
+            return [
+                null,
+                new Response(
+                    json_encode([
+                        'status' => 1,
```
Comment (StefanMavrodiev) [inline src/Upload/ImageUploader.php]: Use _ApiResponseInterface_ ?

File: src/Upload/ImageUploader.php [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+        $extension = $file->getClientOriginalExtension();
+        $extension = mb_strtolower($extension);
+
+        if (!in_array($extension, ['jpeg', 'jpg', 'png', 'gif'])) {
+            return [
+                null,
+                new Response(
+                    json_encode([
+                        'status' => 1,
```
Comment (StefanMavrodiev) [inline src/Upload/ImageUploader.php]: Same here

File: templates/home/home.html.twig [hunk 6]
```diff
@@ -102,69 +103,115 @@
...
-                                track.show();
-                            }
-                            break;
-                        default:
-                            track.show();
+                // places
+                (function (data) {
+                    const responseStatus = data.status;
+
+                    if (responseStatus != 1 && responseStatus != 2) {
```
Comment (StefanMavrodiev) [inline templates/home/home.html.twig]: These are also not friendly.

File: templates/place/view.html.twig [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    <h1>
+        {{ place.name(app.request.locale) }}
+    </h1>
+
+    {% if canEdit %}
+        <a href="{{ path('app_place_edit', {id: place.id}) }}" class="btn btn-info">{{ 'Edit'|trans }}</a>
+    {% endif %}
+
+    {% if app.user %}
+        <label for="file_input" class="btn btn-success mb-0">Add pictures</label>
```
Comment (StefanMavrodiev) [inline templates/place/view.html.twig]: Missing translation

File: src/Form/Type/Place.php [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            ]
+        );
+
+        $builder->add(
+            'type',
+            ChoiceType::class,
+            [
+                'choices' => [
+                    'form.place.type.generic' => \App\Entity\Place::TYPE_GENERIC,
+                    'form.place.type.drinking_water' => \App\Entity\Place::TYPE_DRINKING_FOUNTAIN,
```
[the remark this answers is not in the document]
Comment (angelk) [reply src/Form/Type/Place.php]: changed to drinking fountain

File: src/Contract/Http/ApiResponseInterface.php (hunk 3 above)
Comment (angelk) [reply src/Contract/Http/ApiResponseInterface.php]: Changed to 0

File: src/Upload/ImageUploader.php (hunk 5 above)
Comment (angelk) [reply src/Upload/ImageUploader.php]: fixed

File: src/Upload/ImageUploader.php (hunk 4 above)
Comment (angelk) [reply src/Upload/ImageUploader.php]: fixed

File: templates/place/view.html.twig (hunk 7 above)
Comment (angelk) [reply templates/place/view.html.twig]: added

File: templates/home/home.html.twig (hunk 6 above)
Comment (angelk) [reply templates/home/home.html.twig]: I don't know what is the right way to handle this in JS

**Pull request merged**